src/cordova/setup.action.mjs

```diff
@@ -34,49 +37,173 @@ const WORKING_CORDOVA_OSX_COMMIT = '07e62a53aa6a8a828fd988bc9e884c38c3495a67';
+    case 'browser' + 'debug':
+    default:
+      return cordova.prepare({
+        platforms: ['browser'],
+        save: false,
+      });
   }
+}
 
-  await rmfr('www');
-  await rmfr('platforms');
-  await rmfr('plugins');
+async function androidDebug(verbose) {
+  return cordova.prepare({
+    platforms: ['android'],
+    save: false,
+    verbose,
+  });
+}
 
-  await runAction('www/build', outlinePlatform, `--buildMode=${buildMode}`);
+const makeReplacements = async replacements => {
+  let results = [];
 
-  if (cordovaPlatform === 'osx') {
-    await cordova.platform('add', [`github:apache/cordova-osx#${WORKING_CORDOVA_OSX_COMMIT}`], {save: false});
+  for (const replacement of replacements) {
+    results = [...results, ...(await replace(replacement))];
   }
 
-  if (verbose) {
-    cordova.on('verbose', message => console.debug(`[cordova:verbose] ${message}`));
+  return Promise.resolve(results);
+};
+
+async function androidRelease(version, buildNumber, verbose) {
+  await cordova.prepare({
+    platforms: ['android'],
+    save: false,
+    verbose,
+  });
+
+  const manifestXmlGlob = path.join(getRootDir(), 'platforms', 'android', '**', 'AndroidManifest.xml');
+  const configXmlGlob = path.join(getRootDir(), 'platforms', 'android', '**', 'config.xml');
+
+  return makeReplacements([
+    {
+      files: manifestXmlGlob,
+      from: ['android:versionName="1.0"', 'android:versionName="0.0.0-debug"'],
+      to: `android:versionName="${version}"`,
+    },
+    {
+      files: manifestXmlGlob,
+      from: 'android:versionCode="1"',
+      to: `android:versionCode="${buildNumber}"`,
+    },
+    {
+      files: configXmlGlob,
+      from: 'version="0.0.0-debug"',
+      to: `version="${version}"`,
+    },
+    {
+      files: configXmlGlob,
+      from: 'android-versionCode="1"',
+      to: `android-versionCode="${buildNumber}"`,
+    },
+  ]);
+}
+
+async function appleIosDebug(verbose) {
+  if (os.platform() !== 'darwin') {
+    throw new Error('Building an Apple binary requires xcodebuild and can only be done on MacOS');
+  }
+
+  await cordova.prepare({
+    platforms: ['ios'],
+    save: false,
+    verbose,
+  });
+
+  // TODO(daniellacosse): move this to a cordova hook
+  return spawnStream('rsync', '-avc', 'src/cordova/apple/xcode/ios/', 'platforms/ios/');
+}
+
+async function appleMacOsDebug(verbose) {
+  if (os.platform() !== 'darwin') {
+    throw new Error('Building an Apple binary requires xcodebuild and can only be done on MacOS');
+  }
+
+  await cordova.platform('add', [`github:apache/cordova-osx#${WORKING_CORDOVA_OSX_COMMIT}`], {save: false});
+
+  await cordova.prepare({
+    platforms: ['osx'],
+    save: false,
+    verbose,
+  });
+
+  // TODO(daniellacosse): move this to a cordova hook
+  return spawnStream('rsync', '-avc', 'src/cordova/apple/xcode/macos/', 'platforms/osx/');
+}
+
+const transformAppleXmlFiles = (platform, version, buildNumber) =>
```

Comment:
I'd be more explicit about the intent here.
Perhaps something like `configureVersion` or `setVersions` ?